README.md

```diff
@@ -7,87 +8,127 @@
- Go to your terminal and type the following command
+
+* [Installation](#Installation)
+* [Initialization](#Initialization)
+* [Versioning](#versioning)
+* [URL Generation](#URL-generation)
+* [File Upload](#File-Upload)
+* [File Management](#File-Management)
+* [Utility Functions](#Utility-functions)
+* [Handling errors](#handling-errors)
+* [Support](#Support)
+* [Links](#Links)
+
+## Installation
+
+Go to your terminal and type the following command
+
 ```bash
 pip install imagekitio
 ```
 
 ## Initialization
+
 ```python
 from imagekitio import ImageKit
+
 imagekit = ImageKit(
-    private_key='your private_key',
-    public_key='your public_key',
-    url_endpoint = 'your url_endpoint'
+    private_key='your_private_key',
+    public_key='your_public_key',
+    url_endpoint='your_url_endpoint'
 )
 ```
 
 ## Usage
 
-You can use this Python SDK for 3 different kinds of methods - URL generation, file upload, and file management.
+You can use this Python SDK for 3 different kinds of methods:
+
+- URL generation
+- file upload
+- file management
+
 The usage of the SDK has been explained below.
 
+## Change log
+
+This document presents a list of changes that break existing functionality of previous versions. We try our best to minimize these disruptions, but sometimes they are unavoidable and they will be in major versions.
+
+### Breaking History:
+
+Changes from 2.2.8 -> 3.0.0 are listed below
+
+1. Throw an Error:
+
+**What changed**
+
+- Earlier, in case of failure `error` dict was coming in the `Response` object. Now, an exception is thrown.
+
+**Who is affected?**
+
+- This affects any development in your application that calls APIs from imagekit IO.
+
+**How should I update my code?**
+
+- To avoid failures in an application, you could handle errors as [documented here](#handling-errors)
+
 ## URL generation
 
 **1. Using Image path and image hostname or endpoint**
 
-This method allows you to create a URL using the path where the image exists and the URL
+This method allows you to create a URL using the relative file path where the image exists and the URL
 endpoint(url_endpoint) you want to use to access the image. You can refer to the documentation
 [here](https://docs.imagekit.io/integration/url-endpoints) to read more about URL endpoints
 in ImageKit and the section about [image origins](https://docs.imagekit.io/integration/configure-origin) to understand
 about paths with different kinds of origins.
-
+The file can be an image, video, or any other static file supported by ImageKit.
 
 ```python
 imagekit_url = imagekit.url({
-            "path": "/default-image.jpg",
-            "url_endpoint": "https://ik.imagekit.io/your_imagekit_id/endpoint/",
-            "transformation": [{"height": "300", "width": "400"}],
-        }
+    "path": "/default-image.jpg",
+    "url_endpoint": "https://ik.imagekit.io/your_imagekit_id/endpoint/",
+    "transformation": [{"height": "300", "width": "400", "raw": "ar-4-3,q-40"}],
+}
 )
```

Comment:
Code is not formatted properly. `)` going in new line. Also, all the transformations are written in a single line which is not readable.

Reply:
This is pending. All arrays and transformations. This code should be formatted as below:
```
imagekit_url = imagekit.url({
    "path": "/default-image.jpg",
    "url_endpoint": "https://ik.imagekit.io/your_imagekit_id/endpoint/",
    "transformation": [{
            "height": "300",
            "width": "400",
            "raw": "ar-4-3,q-40"
    }]
})
```